### PR TITLE
IW-1153 | Top articles data for Community Feeds

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/IW-1153
@Wikia/iwing

### CODE REVIEW

File: extensions/wikia/FeedsAndPosts/TopArticles.class.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	const IMAGE_RATIO = 4 / 3;
+
+	public function get() {
+		$cacheTTL = 3600; // an hour
+
+		return \WikiaDataAccess::cache( wfMemcKey( 'feeds-top-articles' ), $cacheTTL, function () {
+			$responseData = [];
+			$mainPageId = \Title::newMainPage()->getArticleID();
+			$insightData =
+				( new \InsightsContext( new \InsightsPopularPagesModel() ) )->fetchData();
```
Comment (rybmat) [inline extensions/wikia/FeedsAndPosts/TopArticles.class.php]: did you consider using this api to fetch articles ordered by pageviews? https://github.com/Wikia/app/blob/9b8b680f9bbf3998dd9f02832b5aaf865998ebc5/includes/wikia/api/ArticlesApiController.class.php#L102
it allows to filter by namespaces

File: includes/wikia/VariablesBase.php [hunk 2]
```diff
@@ -2664,7 +2664,8 @@
  * @see extensions/wikia/FeedsAndPosts
  * @var bool $wgEnableFeedsAndPostsExt
  */
-$wgEnableFeedsAndPostsExt = false;
+// TODO set me back to false
```
Comment (rybmat) [inline includes/wikia/VariablesBase.php:2674]: remember ;)

File: extensions/wikia/FeedsAndPosts/TopArticles.class.php (hunk 1 above)
Comment (latata) [reply extensions/wikia/FeedsAndPosts/TopArticles.class.php]: Changed to ```$data = \F::app()->sendRequest( 'ArticlesApi', 'getTop', $params )->getData();```

File: includes/wikia/VariablesBase.php (hunk 2 above)
Comment (latata) [reply includes/wikia/VariablesBase.php:2674]: I will

**Review by bkoval — APPROVED** [review]

File: extensions/wikia/FeedsAndPosts/Thumbnails.class.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		try {
+			return \VignetteRequest::fromUrl( $url )
+				->zoomCrop()
+				->width( $width )
+				->height( floor( $width / $ratio ) )
+				->url();
+		}
+		catch ( \Exception $exception ) {
+			\Wikia\Logger\WikiaLogger::instance()
+				->warning( "Invalid thumbnail url provided for feeds and posts module", [
```
Comment (rybmat) [inline extensions/wikia/FeedsAndPosts/Thumbnails.class.php]: I'm not sure if this should be warning, this exception will be thrown for example when image from wikipedia is in the article

**Review by rybmat — APPROVED** [review]
one minor thing, generally LGTM

**Pull request merged**